### PR TITLE
Fix: MockSession example and default allocator [172764384]

### PR DESCRIPTION
**Pull request by 678098**
`14NOV2023_17:13:53.982763 48988 8674795648 ERROR ~/blazingmq/src/groups/bmq/bmqa/bmqa_mocksession.cpp 113 UNINITIALIZED_LOGGER_MANAGER [ Description: 'Bad value for argument 'flags' in call to 'bmqa::OpenQueueStatus openQueueSync(QueueId                   *queueId,const bmqt::Uri&           uri,bsls::Types::Uint64        flags,const bmqt::QueueOptions&  options,const bsls::TimeInterval&  timeout)': expected '2', got '10' (~/blazingmq/src/tutorials/subscriptions/consumer.cpp:857)', File: '~/blazingmq/src/tutorials/subscriptions/consumer.cpp:857' ]
FATAL ~/blazingmq/src/groups/bmq/bmqa/bmqa_mocksession.cpp:123 Assertion failed: false
zsh: abort      ./consumer.tsk`

### CODE REVIEW

File: src/groups/bmq/bmqa/bmqa_mocksession.cpp [hunk 1]
```diff
@@ -234,12 +234,14 @@ Event MockSessionUtil::createSessionEvent(
...
 
+    bslma::Allocator* alloc = bslma::Default::allocator(allocator);
+
     Event        event;
     EventImplSp& implPtr = reinterpret_cast<EventImplSp&>(
         static_cast<Event&>(event));
-    implPtr = EventImplSp(new (*allocator)
-                              bmqimp::Event(g_bufferFactory_p, allocator),
-                          allocator);
+    implPtr = EventImplSp(new (*alloc)
```
Comment (678098) [inline src/groups/bmq/bmqa/bmqa_mocksession.cpp]: Fixed some places where `new` was called with allocator pointer not wrapped with `bslma::Default::allocator`. These places were causing crashes when `allocator = 0` passed.

**Review by pniedzielski — DISMISSED** [review]
Two independent changes, right?  `bslma::Default::allocator` and avoiding the hardcoded value for `flags`.

LGTM

**Comment by 678098** [comment]
@pniedzielski re-run clang-format

**Comment by pniedzielski** [comment]
@678098 on src/groups/bmq/bmqa/bmqa_mocksession.cpp as well :)

**Comment by 678098** [comment]
> @678098 on src/groups/bmq/bmqa/bmqa_mocksession.cpp as well :)

I formatted it with clang-format 16.0.6 and it makes a different formatting from CI

**Comment by 678098** [comment]
https://github.com/llvm/llvm-project/issues/54703
Looks like a fixed bug

**Review by pniedzielski — APPROVED** [review]
There's a failure in the CI that doesn't appear to have to do with this PR (it's failing on `main` as well, and doesn't appear on local builds).  This PR is good to go, but we'll investigate that issue separately.

**Pull request merged**